### PR TITLE
Add support for script nonce attributes

### LINKED ISSUE #430: Custom attributes in `ReactInitJavaScript` script tag
Same as the now closed issue #334 

Like the aforementioned post, I'm about to implement CSP on a site using ReactJS.NET.
For the inline script created by ReactInitJavaScript to work i need to add a nonce attribute to the script tag with a value that matches one in my CSP headers.

Currently I have created my own helper, similar in functionality to ReactInitJavaScript. 
However, seeing as this configuration might become more and more common as time goes on, I think it might make sense to integrate into the project. 

Do you agree? 
And yes I'd be willing to submit a pull request :)

### PR DESCRIPTION
**Pull request by dustinsoftware**
Fixes #430, #482, #334

### CODE REVIEW

**Comment by dustinsoftware** [comment]
@Daniel15 this is good to go, but I'd like another set of eyes to give it a quick look before it gets merged 😄

**Comment by dustinsoftware** [comment]
Merging this so it doesn't get stale, let me know if needs modification before shipping. Tested on Mvc4 and CoreMvc.

File: src/React.Core/IReactEnvironment.cs [hunk 1]
```diff
@@ -110,5 +109,10 @@ public interface IReactEnvironment
 		/// Returns the currently held JS engine to the pool. (no-op if engine pooling is disabled)
 		/// </summary>
 		void ReturnEngineToPool();
+
+		/// <summary>
+		/// Gets the site-wide configuration.
+		/// </summary>
+		IReactSiteConfiguration Configuration { get; }
```
Comment (Daniel15) [inline src/React.Core/IReactEnvironment.cs]: Isn't there already a getter for this? `ReactSiteConfiguration.Configuration` or `.Current` or something like that?

**Comment by dustinsoftware** [comment]
I can check later, the prop already existed on the implementation but not
the interface.. don’t remember seeing another one I could use

**Pull request merged**